hpx/runtime/threads/detail/scheduling_loop.hpp

```diff
@@ -302,6 +302,25 @@ namespace hpx { namespace threads { namespace detail
         thread_data* thrd = nullptr;
         thread_data* next_thrd = nullptr;
 
+        thread_init_data background_init(
+            [&](thread_state_ex_enum) -> thread_result_type
+            {
+                while(true)
+                {
```

Comment:
I think this could be changed to:

    while(running) { ... }

where `running` is captured from the enclosing frame (by reference).